### PR TITLE
catppuccin-sddm: init at 1.0.0

### PR DESCRIPTION
**Pull request by ghost**
## Description of changes

Adding package `catppuccin-sddm` on nixpkgs, since we already have some other catppuccin official themes (gtk, plasma, icons etc...).

I've added options to the derivation to control the theme file (flavor, theme.conf options...).

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by keenanweaver** [comment]
Result of `nixpkgs-review pr 296682` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>catppuccin-sddm</li>
  </ul>
</details>

**Review by keenanweaver — APPROVED** [review]
Working well for me. This would also close https://github.com/NixOS/nixpkgs/pull/279136

**Comment by khaneliman** [comment]
> Working well for me. This would also close #279136

https://github.com/NixOS/nixpkgs/pull/255808 as well

File: pkgs/data/themes/catppuccin-sddm/default.nix [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    substituteInPlace $configFile \
+        --replace 'Font="Noto Sans"' 'Font="${font}"'
+
+    substituteInPlace $configFile \
+        --replace 'FontSize=9' 'FontSize=${fontSize}'
+
+    ${lib.optionalString (background != null) ''
+      substituteInPlace $configFile \
+      --replace 'Background="backgrounds/wall.jpg"' 'Background="${background}"'
+      substituteInPlace $configFile \
```
Comment (SuperSandro2000) [inline pkgs/data/themes/catppuccin-sddm/default.nix:48]:
```suggestion
      --replace 'Background="backgrounds/wall.jpg"' 'Background="${background}"' \
```

File: pkgs/data/themes/catppuccin-sddm/default.nix [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  installPhase = ''
+    mkdir -p "$out/share/sddm/themes/"
+    cp -r src/catppuccin-${flavor} "$out/share/sddm/themes/catppuccin-${flavor}"
+
+    configFile=$out/share/sddm/themes/catppuccin-${flavor}/theme.conf
+
+    substituteInPlace $configFile \
+        --replace 'Font="Noto Sans"' 'Font="${font}"'
+
+    substituteInPlace $configFile \
```
Comment (SuperSandro2000) [inline pkgs/data/themes/catppuccin-sddm/default.nix:42]:
```suggestion
        --replace-fail 'Font="Noto Sans"' 'Font="${font}"' \
```

Please use the new --replace-fail argument everywhere

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -5679,6 +5679,11 @@
     githubId = 1365692;
     name = "Will Fancher";
   };
+  elysasrc = {
+    name = "Elysa";
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:5683]: Would you mind adding an email or matrix?

File: pkgs/data/themes/catppuccin-sddm/default.nix [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  qtquickcontrols2,
+  flavor ? "mocha",
+  font ? "Noto Sans",
+  fontSize ? "9",
+  background ? null,
+  loginBackground ? false,
+}:
+stdenvNoCC.mkDerivation {
+  pname = "catppuccin-sddm";
+  version = "main-2024-03-17";
```
Comment (jopejoe1) [inline pkgs/data/themes/catppuccin-sddm/default.nix:16]: Version 1.0.0 just got released https://github.com/catppuccin/sddm/releases

File: maintainers/maintainer-list.nix (hunk 3 above)
Comment (ElysaSrc) [reply maintainers/maintainer-list.nix:5683]: Added my email.

File: pkgs/data/themes/catppuccin-sddm/default.nix (hunk 4 above)
Comment (ElysaSrc) [reply pkgs/data/themes/catppuccin-sddm/default.nix]: Updated to 1.0.0, thank you for pointing this out.

**Comment by ElysaSrc** [comment]
Force pushed the branch:
- Bumped to v1.0.0
- Using `--replace-fail`
- Cleaned up redundant `substituteInPlace`

File: pkgs/data/themes/catppuccin-sddm/default.nix [hunk 5]
```diff
@@ -0,0 +1,78 @@
+{
+  lib,
+  stdenvNoCC,
+  fetchFromGitHub,
+  just,
+  qtgraphicaleffects,
```
Comment (isabelroses) [inline pkgs/data/themes/catppuccin-sddm/default.nix:6]: Hello, I am the maintainer of catppuccin/sddm. We don't have `qtgraphicaleffects` as a dependency anymore.

Comment (ElysaSrc) [reply pkgs/data/themes/catppuccin-sddm/default.nix:6]: Thank you ! I'm going to fix this today.

**Review by isabelroses — APPROVED** [review]

**Comment by ElysaSrc** [comment]
Is there any issue remaining preventing this PR to be merged ?

**Comment by ElysaSrc** [comment]
Updated the branch, using kdePackages.callPackage instead of the libsForQt5.callPackage

**Review by Aleksanaa — COMMENTED** [review]
You have to add `runHook` if you override phases. See https://nixos.org/manual/nixpkgs/unstable/#sec-stdenv-phases

Here are `buildPhase` and `installPhase`

File: pkgs/data/themes/catppuccin-sddm/default.nix [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    owner = "catppuccin";
+    repo = "sddm";
+    rev = "v${version}";
+    hash = "sha256-SdpkuonPLgCgajW99AzJaR8uvdCPi4MdIxS5eB+Q9WQ=";
+  };
+
+  dontWrapQtApps = true;
+
+  buildInputs = [
+    just
```
Comment (Aleksanaa) [inline pkgs/data/themes/catppuccin-sddm/default.nix:27]: `just` should be in `nativeBuildInputs`.

Comment (ElysaSrc) [reply pkgs/data/themes/catppuccin-sddm/default.nix:27]: Thanks, modified it !

**Comment by Sigmanificient** [comment]
Can you re-fetch master and rebase? Eval is failing and was fixed by #305609.

**Comment by ElysaSrc** [comment]
@Sigmanificient That should be ok

**Review by pluiedev — CHANGES_REQUESTED** [review]
Shouldn't all new packages be using the `pkgs/by-name` convention, per [RFC 140](https://github.com/NixOS/rfcs/pull/140)?

**Comment by ElysaSrc** [comment]
@pluiedev I'm sorry, I checked the RFC but didn't understand what I'm supposed to do ? Could you point me in a direction ? Thanks !

**Comment by pluiedev** [comment]
> @pluiedev I'm sorry, I checked the RFC but didn't understand what I'm supposed to do ? Could you point me in a direction ? Thanks !

Move `pkgs/data/themes/catppuccin-sddm/default.nix` to `pkgs/by-name/ca/catppuccin-sddm/package.nix`, and remove the addition to `pkgs/top-level/all-packages.nix`

**Review by pluiedev — APPROVED** [review]

**Comment by ElysaSrc** [comment]
> > @pluiedev I'm sorry, I checked the RFC but didn't understand what I'm supposed to do ? Could you point me in a direction ? Thanks !
> 
> Move `pkgs/data/themes/catppuccin-sddm/default.nix` to `pkgs/by-name/ca/catppuccin-sddm/package.nix`, and remove the addition to `pkgs/top-level/all-packages.nix`

Sorry for the delay, made the changes, rebased on more recent master.

**Review by pluiedev — APPROVED** [review]

**Comment by ribru17** [comment]
Thank you so much for this! Just wanted to note that when using this theme the `settings.Theme.CursorTheme` option for `sddm` is now ignored for me. Must those values be set in `catppuccin-sddm`'s `sddm.conf`?

**Pull request merged**